components/engine_rocks/src/checkpoint.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::path::Path;
+
+use engine_traits::{Checkpointable, Checkpointer, Result};
+
+use crate::{r2e, RocksEngine};
+
+impl Checkpointable for RocksEngine {
```

Comment:
RocksDB interface is too cumbersome, the overhead of `new_checkpointer` is negligible. Let's just expose only one trait and one interface `create_checkpoint`.

Reply:
How about your suggestions about this? @BusyJay

Reply:
I prefer to keep align with rocksdb's API.